SteamKit2/SteamKit2/Steam/Discovery/SmartCMServerList.cs

```diff
@@ -264,7 +233,8 @@ private IPEndPoint GetNextServerCandidateInternal()
                 ResetOldScores();
 
                 var serverInfo = servers
-                    .Select( (s, index) => new { EndPoint = s.EndPoint, IsBad = s.LastBadConnectionDateTimeUtc.HasValue, Index = index } )
+                    .Where( s => (s.Record.ProtocolTypes & supportedProtocolTypes) > 0 )
```

Comment:
`HasFlag` is a slower check at runtime, at least on .NET Framework. How do you feel about an extension method, so we have both clarity and speed?